docs/grammar/Solidity.g4

```diff
@@ -0,0 +1,489 @@
+]
+:
+    Modifier name=identifier
+    (LParen (arguments=parameterList)? RParen)?
+    (
+        {!$virtualSet}? Virtual {$virtualSet = true;}
+        | {!$overrideSpecifierSet}? overrideSpecifier {$overrideSpecifierSet = true;}
+    )*
+    (Semicolon | body=block);
+
+/**
+ * Definitions of the special fallback and receive functions.
+ */
+fallbackReceiveFunctionDefinition
+locals[
+    boolean visibilitySet = false,
+    boolean mutabilitySet = false,
+    boolean virtualSet = false,
+    boolean overrideSpecifierSet = false
+]
+:
+	kind=(Fallback | Receive) LParen RParen
+	(
+	    {!$visibilitySet}? visibility {$visibilitySet = true;}
+	    | {!$mutabilitySet}? stateMutability {$mutabilitySet = true;}
+	    | modifierInvocation
+	    | {!$virtualSet}? Virtual {$virtualSet = true;}
+	    | {!$overrideSpecifierSet}? overrideSpecifier {$overrideSpecifierSet = true;}
+	 )*
+	(Semicolon | body=block);
+
+/**
+ * Definition of a struct. Can occur at top-level within a source unit or within a contract, library or interface.
+ */
+structDefinition: Struct name=identifier LBrace members=structMember+ RBrace;
+/**
+ * The declaration of a named struct member.
+ */
+structMember: type=typeName name=identifier Semicolon;
+/**
+ * Definition of an enum. Can occur at top-level within a source unit or within a contract, library or interface.
+ */
+enumDefinition:	Enum name=identifier LBrace enumValues+=identifier (Comma enumValues+=identifier)* RBrace;
+
+/**
+ * The declaration of a state variable.
+ */
+stateVariableDeclaration
+locals [boolean constantnessSet = false, boolean visibilitySet = false, boolean overrideSpecifierSet = false]
+:
+	type=typeName
+	(
+	    {!$visibilitySet}? Public {$visibilitySet = true;}
+	    | {!$visibilitySet}? Private {$visibilitySet = true;}
+	    | {!$visibilitySet}? Internal {$visibilitySet = true;}
+	    | {!$constantnessSet}? Constant {$constantnessSet = true;}
+	    | {!$overrideSpecifierSet}? overrideSpecifier {$overrideSpecifierSet = true;}
+	    | {!$constantnessSet}? Immutable {$constantnessSet = true;}
+	)*
+	name=identifier
+	(Assign initialValue=expression)?
+	Semicolon;
+
+/**
+ * Parameter of an event.
+ */
+eventParameter: type=typeName Indexed? name=identifier?;
+/**
+ * Definition of an event. Can occur in contracts, libraries or interfaces.
+ */
+eventDefinition:
+    Event name=identifier
+    LParen (parameters+=eventParameter (Comma parameters+=eventParameter)*)? RParen
+    Anonymous?
+    Semicolon;
+
+/**
+ * Using directive to bind library functions to types.
+ * Can occur within contracts and libraries.
+ */
+usingDirective: Using userDefinedTypeName For (Mul | typeName) Semicolon;
+/**
+ * A type name can either be an elementary type, a function type, a mapping type, a user-defined type
+ * (e.g. a contract or struct) or an array type.
+ */
+typeName: elementaryTypeName[true] | functionTypeName | mappingType | userDefinedTypeName | typeName LBrack expression? RBrack;
```

Comment:
What's the last one related to?

Reply:
The last one is array types ``uint256[]`` or ``uint256[42]``, resp. ``uint256[3 + 3]``.